lib/libc/common/source/stdlib/malloc.c

```diff
@@ -153,6 +153,14 @@ void *aligned_alloc(size_t alignment, size_t size)
 	return ret;
 }
 
+#ifdef CONFIG_CPP
```

Comment:
Make this `#if defined(CONFIG_CPP) && defined(__GNUC__)`

Reply:
that will hit clang as well though, right?

Reply:
done; we can adjust for clang if necessary.